thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/api/detection/AnomalyDetectionResource.java

```diff
@@ -0,0 +1,756 @@
+
+    return detectionConfigDTO;
+  }
+
+  TaskDTO generateTaskConfig(long detectionConfigId, long start, long end)
+      throws JsonProcessingException {
+    TaskDTO taskDTO = new TaskDTO();
+    taskDTO.setJobName(TaskConstants.TaskType.DETECTION.toString() + "_" + detectionConfigId);
+    taskDTO.setStatus(TaskConstants.TaskStatus.WAITING);
+    taskDTO.setTaskType(TaskConstants.TaskType.DETECTION_ONLINE);
+    DetectionPipelineTaskInfo taskInfo =
+        new DetectionPipelineTaskInfo(detectionConfigId, start, end);
+    String taskInfoJson = objectMapper.writeValueAsString(taskInfo);
+    taskDTO.setTaskInfo(taskInfoJson);
+
+    taskDAO.save(taskDTO);
+    LOG.info("Created task: {}", taskDTO);
+
+    return taskDTO;
+  }
+
+  private TaskDTO pollingTask(long taskId) {
+    long startTime = System.currentTimeMillis();
+    TaskDTO taskDTO;
+
+    // Timeout mechanism will be handled by worker thread in the controller
+    while (true) {
+      taskDTO = taskDAO.findById(taskId);
+
+      LOG.info("Polling task : " + taskDTO);
+
+      TaskConstants.TaskStatus taskStatus = taskDTO.getStatus();
+      if (!taskStatus.equals(TaskConstants.TaskStatus.WAITING) && !taskStatus
+          .equals(TaskConstants.TaskStatus.RUNNING)) {
+        LOG.info("Polling finished ({}ms). Task status: {}", System.currentTimeMillis() - startTime,
+            taskStatus);
+        break;
+      }
+
+      try {
+        TimeUnit.SECONDS.sleep(POLLING_SLEEP_TIME);
+      } catch (InterruptedException e) {
+        Log.warn("Interrupted during polling sleep");
+        break;
+      }
+    }
+
+    return taskDTO;
+  }
+
+  private List<AnomalySummary> getAnomalies(long start, long end, String metric, String dataset) {
+    List<AnomalySummary> anomalies =
+        this.userDashboardResource.queryAnomalies(start, end, null, null, metric,
+            dataset, null, false, null);
+
+    LOG.info("Successfully returned " + anomalies.size() + " anomalies.");
+    return anomalies;
+  }
+
+  private void cleanStates(List<AnomalySummary> anomalies, TaskDTO taskDTO,
+      MetricConfigDTO metricConfigDTO, DatasetConfigDTO datasetConfigDTO,
+      DetectionConfigDTO detectionConfigDTO) {
+    if (anomalies != null) {
+      for (AnomalySummary anomaly : anomalies) {
+        anomalyDAO.deleteById(anomaly.getId());
+        LOG.info("Deleted anomaly with id: {}", anomaly.getId());
+      }
+    }
+
+    if (datasetConfigDTO != null) {
+      datasetConfigDAO.delete(datasetConfigDTO);
+      LOG.info("Deleted dataset: {}", datasetConfigDTO);
+    }
+
+    if (metricConfigDTO != null) {
+      metricConfigDAO.delete(metricConfigDTO);
+      LOG.info("Deleted metric: {}", metricConfigDTO);
+    }
+
+    if (detectionConfigDTO != null) {
+      detectionConfigDAO.delete(detectionConfigDTO);
+      LOG.info("Deleted detection: {}", detectionConfigDTO);
+    }
+
+    if (taskDTO != null) {
+      taskDAO.delete(taskDTO);
```

Comment:
no need to delete the taskDTO. it will be cleaned up periodically by the monitor task.